oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java

```diff
@@ -220,6 +222,35 @@ public String getElasticKeyword(String propertyName) {
         return field;
     }
 
+    public String getElasticTextField(String propertyName) {
+        List<PropertyDefinition> propertyDefinitions = propertiesByName.get(propertyName);
+        if (propertyDefinitions == null) {
+            return propertyName;
+        }
+
+        Type<?> type = null;
+        for (PropertyDefinition pd : propertyDefinitions) {
+            type = Type.fromTag(pd.getType(), false);
+        }
+
+        if (isAnalyzed(propertyDefinitions)) {
+            // The full text index for String properties is <propertyName>, while for non-string properties is part of
+            // the multi-field and is called <propertyName>.text
+            if (Type.BINARY.equals(type) ||
```

Comment:
Isn't BINARY also indexed as text?

Reply:
In Elastic, Binary is a Base64 encoded string, which by default is not indexed.
https://www.elastic.co/guide/en/elasticsearch/reference/7.17/binary.html

I have not changed the way the JCR binary type is mapped, even before this PR, the mapping logic handles Binary, Long, Double, Date and Boolean all the in the same way, and text in a different way. But Binary does seem to require special treatment, as there are no good use cases for it to be indexed. That is, are there any situations where it may make sense to run either term-level or full-text queries on Binary fields?